### DISCUSSION TITLE
'Module not specified' error while testing the app

### DISCUSSION THREAD

**Discussion by dmdbilal** [oppia/oppia-android#4816] [Q&A (Other)] [answered]
I'm getting this error while testing the ProfileChooserActivityTest and ProfileChooserFragmentTest.
I've tried with clean and rebuilding the project but still i'm getting the same issue.
![Screenshot 2022-12-22 203126](https://user-images.githubusercontent.com/91405690/210126315-45e6c416-64e1-4792-acdb-c13c2144161c.png)

**Comment by MohitGupta121** [comment]
@YesShubham01 Which Android Studio version you are using?
Recommended version for oppia-android to run Tests is - **Android Studio Bumblebee (Patch 3)** as it's also works for me and also you refer to this [discussion](https://github.com/oppia/oppia-android/discussions/4432).



If it fixed by this answer please mark this as answer or if you find any other way that work for you please share and mark that as answer. So that this helps others in community.

**Reply by dmdbilal** [reply]
I've tried in Android Studio Bumblebee(Patch 3). And i cleaned and rebuild the project but still i'm getting the same issue.
![Screenshot 2023-01-01 113904](https://user-images.githubusercontent.com/91405690/210162509-bc967050-3e72-4e11-a195-81d5ee141d39.png)

**Reply by MohitGupta121** [reply]
@dmdbilal can you please try invalidate cache and restart. 
As I also getting the same error and these steps fixed my issue.

**Reply by dmdbilal** [reply]
First I've tried invalidate cache and restart. And then i tried rebuilding project. Still i'm getting the same issue.

**Reply by MohitGupta121** [reply]
Which Os you using?

**Reply by dmdbilal** [reply]
Windows 11

**Reply by dmdbilal** [reply]
What should i do now ?

**Reply by MohitGupta121** [reply]
Can you please follow this [wiki](https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing#overview)
Hope this will helps something.

**Reply by dmdbilal** [reply]
I've also tried testing in Android Instruentation Test and Robolectric tests both are not working for this issue.

**Reply by MohitGupta121** [reply]
Yes no one works, I think there is some issue in setup itself you please try it again cloning and build project. Make sure have have suitable Android Studio version Android Studio Bumblebee (Patch 3).

**Reply by MohitGupta121** [reply]
Hey @dmdbilal if your issue is resolved please mark this discussion is resolved (marking as answered that comment) so that it's helpful to others in the community.
And if you need help then also feel free to ask.

**Comment by Rd4dev** [comment]
Following @MohitGupta121 [instructions](https://github.com/oppia/oppia-android/discussions/4816#discussioncomment-4560900), 
Using **Android Studio Bumblebee (Patch 3)**, I was able to resolve the `'module not specified'` error. 

![ProfileAcivityTest](https://github.com/oppia/oppia-android/assets/122200035/9cbe9563-67c0-4a55-9138-0e60a362f25d)

@dmdbilal, if you're still experiencing issues with Android Bumble Bee Patch 3, feel free to share your results here, and I'll do my best to assist you.

**Comment by dmdbilal** [comment]
For the last 4 weeks I had my university exams, that's why i can't be active. 

Btw I think there might be some issues with my Android Studio version, so I unninstalled and installed again the Android Studio Bumblebee(Path 3). After that the problem was resolved. Now i can make JUnit tests.

But now when I run the test, i'm getting the below error.

![Screenshot 2024-02-02 200608](https://github.com/oppia/oppia-android/assets/91405690/b3e224cc-a72e-4f65-a77e-0e953ea83a12)

**Reply by Rd4dev** [reply]
Hi @dmdbilal, glad you resolved the `module not found error`. Could you please confirm if you were able to build the project successfully after invalidating caches? This will help us ensure if the issue you're facing is specific to running tests.

**Reply by dmdbilal** [reply]
I'm getting the same error after invalidating caches.

**Reply by Rd4dev** [reply, marked as answer]
@dmdbilal, could you take a moment to verify if the SDK location is correctly configured in the **local.properties** file? ([Video for reference](https://www.youtube.com/watch?v=3fsr2-ti39w)). If the issue persists, it might be beneficial to regenerate the file and see if that resolves the error. 

_Additionally, I think it would be a good idea to create a new issue to address this matter more effectively._
